File: DESIGN-CSharp/Strategy/Adventure.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;


namespace Strategy
{
    class Adventure
    {
        FlightStrategy flightStrategy;

        public void attack() 
        {
            if (flightStrategy == null) 
            {
                flightStrategy = new NormalAttack();
            }
            flightStrategy.execute();
        }

        public void choiceStrategy(FlightStrategy strategy)
        {
            this.flightStrategy = strategy;
        }
    }
}

File: BehavioralPatterns/Strategy/Adventure.py
from FlightStrategy import NormalAttack, UseSkill, UseItem

class Adventure:
    def __init__(self):
        self.flightStrategy = None

    def attack(self):
        if self.flightStrategy is None:
            self.flightStrategy = NormalAttack()
        self.flightStrategy.execute()

    def chooseStrategy(self, strategy):
        self.flightStrategy = strategy